### PR TITLE
feature/ocn_da_diag_table

### PR DESCRIPTION
**Pull request by hyunchul386**
## Description

To add ocean history files by diag_table, which save hourly fields for ocean DA at the native grid in MOM6.
This PD does not change the answers

git diff --name-status develop
A       parm/soca/mom6/diag_table

### Issue(s) addressed
- fixes #365 ( https://github.com/NOAA-EMC/godas/issues/354 )

## Testing

After run check the hourly files of ocn_da_yyyy_mm_dd_hh.nc, 
which save variables of geolon, geolat, SSH, MDL, Temp, Salt, u, v, h 

## Dependencies

Using for ocean DA

### CODE REVIEW

File: parm/soca/mom6/diag_table [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+"fv3_history",    0,  "hours",  1,  "hours",  "time"
+"fv3_history2d",  0,  "hours",  1,  "hours",  "time"
+######################
+"ocn_da%4yr%2mo%2dy%2hr",      0,  "hours", 1, "hours", "time", 1, "hours", "1901 1 1 0 0 0"
+##############################################
+# static fields
+  "ocean_model", "geolon",      "geolon",      "ocn_da%4yr%2mo%2dy%2hr", "all", .false., "none", 2
+  "ocean_model", "geolat",      "geolat",      "ocn_da%4yr%2mo%2dy%2hr", "all", .false., "none", 2
+# ocean output TSUV and others
+ "ocean_model", "SSH",       "SSH",      "ocn_da%4yr%2mo%2dy%2hr","all",.true.,"none",2
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: `SSH` should map to `ave_ssh`

File: parm/soca/mom6/diag_table [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+#"ocean_model", "SST",       "SST",      "ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+#"ocean_model", "SSS",       "SSS",      "ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: Remove `SSS` and `SST`, neither are needed.

File: parm/soca/mom6/diag_table [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+ "ocean_model", "MLD_0125",  "MLD",      "ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+#===============================================
+
+
+# Z-Space Fields Provided for CMIP6 (CMOR Names):
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: Comment doesn't match why we're saving these fileds. Remove or update the comment.

File: parm/soca/mom6/diag_table [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+#===============================================
+ "ocean_model","u","u"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","v","v"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","salt","Salt"      ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","temp","Temp"      ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","h","h"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+
+###
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: None of the stuff below is needed

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Thankls @hyunchul386 . I know I'm moving the target, my apologizes, but since we're only putting this diag file in this repo as a reference as to what to add to the current diag table. Could you remove what's not relevant to `ocn_da` please.

File: parm/soca/mom6/diag_table (hunk 4 above)
Comment (hyunchul386) [reply parm/soca/mom6/diag_table]: @guillaumevernieres  Sure, diag_table has been updated

File: parm/soca/mom6/diag_table [hunk 5]
```diff
@@ -0,0 +1,58 @@
+20210322.06Z.C48.64bit.non-mono
+2021 03 22 06 0 0
+
+"fv3_history",    0,  "hours",  1,  "hours",  "time"
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: `fv3` stuff that I don't care about ... Remove, not needed

File: parm/soca/mom6/diag_table [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+######################
+"ocn_da%4yr%2mo%2dy%2hr",      0,  "hours", 1, "hours", "time", 1, "hours", "1901 1 1 0 0 0"
+##############################################
+# static fields
+  "ocean_model", "geolon",      "geolon",      "ocn_da%4yr%2mo%2dy%2hr", "all", .false., "none", 2
+  "ocean_model", "geolat",      "geolat",      "ocn_da%4yr%2mo%2dy%2hr", "all", .false., "none", 2
+# ocean output TSUV and others
+ "ocean_model", "SSH",       "ave_ssh",      "ocn_da%4yr%2mo%2dy%2hr","all",.true.,"none",2
+ "ocean_model", "MLD_0125",  "MLD",      "ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+#===============================================
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table:14]: Not sure why we need a separation here.

File: parm/soca/mom6/diag_table [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+#===============================================
+ "ocean_model","u","u"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","v","v"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","salt","Salt"      ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","temp","Temp"      ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+ "ocean_model","h","h"            ,"ocn_da%4yr%2mo%2dy%2hr","all",.false.,"none",2
+
+###
+#=============================================================================================
```
Comment (guillaumevernieres) [inline parm/soca/mom6/diag_table]: Whatever is below will be in the diag_table that we will modify in the ufs config, I would say that needs to go as well.

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
A bit of "code" tidy/erasing and that should be good to go @hyunchul386

File: parm/soca/mom6/diag_table (hunk 7 above)
Comment (hyunchul386) [reply parm/soca/mom6/diag_table]: The table was updated (PR2)

**Review by guillaumevernieres — APPROVED** [review]
Thanks @hyunchul386

**Comment by hyunchul386** [comment]
Followings are comparisons of T and S between restart files and history files
As shown, the difference of temperature between restart and history files are less than 10**-6 Celsius Degree at the surface layer and the difference gradually decreases with depth. The salinity differences are similar to one of temperature,
less than 1-**-7 psu at the first layer, decreasing with depth.
The difference may be due to the precision differences between single in history file and double in restart file. 


Temp_diff_1
Salt_diff_1
Temp_diff_2
Salt_diff_2

**Pull request merged**